build-system/tasks/storybook/index.js

```diff
@@ -64,23 +67,23 @@ function buildEnv(env) {
         'See https://github.com/ampproject/storybook-addon-amp/issues/57'
     );
     return;
+    if (env === 'amp' && isPullRequestBuild()) {
+      // Allows PR deploys to reference built binaries.
+      writeFileSync(
+        `${envConfigDir(env)}/preview.js`,
+        // If you change this JS template, make sure to JSON.stringify every
+        // dynamic value. This prevents XSS and other types of garbling.
+        `// DO NOT${' '}SUBMIT.
+         // This preview.js file was generated for a specific PR build.
+         export const parameters = (${JSON.stringify({
+           ampBaseUrlOptions: [`${getBaseUrl()}/dist`],
+         })});`
+      );
```

Comment:
Looking at this, I had a hard time following the nested backticks and `${...}`. Suggestion for readability/making it less likely someone like me breaks this somehow:

```suggestion
      const parameters = {
         ampBaseUrlOptions: [`${getBaseUrl()}/dist`],
       };
      const previewFileContents = [
          // eslint-disable-next-line local/no-forbidden-terms
          '// DO NOT SUBMIT',
          '// This preview.js file was generated for a specific PR build.',
          // JSON.stringify here. prevents XSS and other types of garbling.
          `export const parameters = (${JSON.stringify(parameters)});`,
        ].join('\n');
      writeFileSync(`${envConfigDir(env)}/preview.js`, previewFileContents);
```

It's a bit bulkier (esp. splitting params/contents into separate vars) but might be more resilient to future edits, especially to `parameters`

Also, it avoids extra leading whitespace on lines 2 and 3 in the preview. Which doesn't matter AT ALL but is still slightly satisfying 🙃